### PR TITLE
Standardize abstract_charm.py file with the K8s charm

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
The `abstract_charm.py` has diverged with the K8s charm following in support of HACluster charm integration + `expose-external` config in the K8s charm.

## Solution
Standardize the files

Counterpart PR in K8s charm: https://github.com/canonical/mysql-router-k8s-operator/pull/328

### CODE REVIEW

**Review by paulomach — APPROVED** [review]
lgtm

**Comment by taurus-forever** [comment]
> please wait for [canonical/mysql-router-k8s-operator#328](https://github.com/canonical/mysql-router-k8s-operator/pull/328) merge before merging this pr

I am pending decision in K8s before reviewing VM one.

**Comment by paulomach** [comment]
> > > please wait for [canonical/mysql-router-k8s-operator#328](https://github.com/canonical/mysql-router-k8s-operator/pull/328) merge before merging this pr
> > 
> > 
> > I am pending decision in K8s before reviewing VM one.
> 
> @paulomach [K8s](https://github.com/canonical/mysql-router-k8s-operator/pull/328) got merged yesterday. Should this follow?

@taurus-forever no. It's currently out of sync. I rather wait for @shayancanonical to resync it.

**Review by sinclert-canonical — APPROVED** [review]

File: src/abstract_charm.py [hunk 1]
```diff
@@ -341,14 +351,7 @@ def reconcile(self, event=None) -> None:  # noqa: C901
                         exposed_read_only_endpoint=self._exposed_read_only_endpoint,
                         shell=workload_.shell,
                     )
-                    # _ha_cluster only assigned a value in machine charms
-                    if self._ha_cluster:
-                        self._database_provides.update_endpoints(
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply src/abstract_charm.py:346]: outdated now. we introduce a new abstract method `self._update_endpoints()` to update endpoints differently in vm and k8s

**Pull request merged**